### DISCUSSION TITLE
How do I change my local repository `master` branch to `main`?

### DISCUSSION THREAD

**Discussion by glemieux** [NGEET/fates#944] [Q&A] [answered]
The FATES team changed the name of the master branch to main on November 22, 2022. While GitHub automatically handles the retargeting of names behind the scenes, users need to manually update their local branches to point to the new main branch.  The comment below provides the instructions.

**Comment by glemieux** [comment, marked as answer]
TL;DR:
```
git branch -m master main
git fetch <repo remote name>
git branch -u <repo remote name>/main main
git remote set-head <repo remote name> -a
```

Step-by-step explanation:

The following assumes that the user has cloned the NGEET repository and have not changed the `master` branch name already.  The `<repo remote name>` below defaults to `origin` unless you have changed it to something different (using `git remote rename` command).  To check the name of the remote associated with NGEET, run the `git remote -v` command and look for the name associated with the `NGEET/fates.git` address.

1.  Rename your local `master` branch to `main`: `git branch -m master main`
2. Get the latest updates from the NGEET repository: `git fetch <repo remote name>`
3. Set the local `main` branch to track the remote `main` branch: `git branch -u <repo remote name>/main main`
4. Set your local default branch for the remote repository: `git remote set-head <repo remote name> -a`

For personal forks of the NGEET repository, you may also update your personal `master` branch to `main`.  To do this you will need to go to GitHub to rename your branch and then follow the above instructions for the local clone of your personal repository.  See the [Github documentation](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/renaming-a-branch) on renaming branches for details.  When you update the branch on Github, it will provide multiple reminders about how to rename your local repo branch as seen in the screen shot below:
![Screen Shot 2022-11-22 at 10 54 01 AM](https://user-images.githubusercontent.com/7565064/203434836-26c58eef-2ba7-46a8-bba4-fc8056b716d7.png)

Reference issue: https://github.com/NGEET/fates/issues/664
